regression: Unable to create power instance using catalog image

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Terraform CLI and Terraform IBM Provider Version

<!--- Please run `terraform -v` to show the Terraform core version and provider version(s). If you are not running the latest version of Terraform or the provider, please upgrade because your issue may have already been fixed. [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions). --->


### Affected Resource(s)

<!--- Please list the affected resources and data sources. --->

* ibm_pi_instance

### Terraform Configuration Files

<!--- Information about code formatting: https://help.github.com/articles/basic-writing-and-formatting-syntax/#quoting-code --->

Please include all Terraform configurations required to reproduce the bug. Bug reports without a functional reproduction may be closed without investigation.

```hcl

locals {
  centos_img_storaget_lon04 = tolist([for obj in data.ibm_pi_catalog_images.catalog_lon04.images: obj.storage_type if strcontains("CentOS-Stream-8", obj.name)])[0] 
  centos_img_id_lon04 = tolist([for obj in data.ibm_pi_catalog_images.catalog_lon04.images: obj.image_id if strcontains("CentOS-Stream-8", obj.name)])[0]
}

resource "ibm_pi_instance" "centos_instances_lon04" {
  provider             = ibm.powervslon04
  count                = 1
  pi_memory            = "8" 
  pi_processors        = "0.5"
  pi_instance_name     = "rdr-cln-centos-lon04"
  pi_proc_type         = "shared"
  pi_image_id          = local.centos_img_id_lon04
  pi_storage_type      = local.centos_img_storaget_lon04
  pi_network           { network_id = ibm_pi_dhcp.dhcp_lon04.id}
  pi_key_pair_name     = data.ibm_pi_key.key_lon04.id
  pi_sys_type          = "s922"
  pi_cloud_instance_id = var.cloud_inst_id_lon04
  pi_health_status     = "WARNING"
}


```

### Debug Output

https://gist.github.com/clnperez/5ef202b04ad45a79d220c716c95e74c8

### Panic Output

n/a

### Expected Behavior

the instance should have created successfully. i have been able to use the catalog images with terraform in the past

### Actual Behavior

it seems to be only looking for images in the service instance to verify, and fails because it can't find it there


### Steps to Reproduce

<!--- Please list the steps required to reproduce the issue. --->

1. `terraform apply`


### Important Factoids

<!--- Are there anything atypical about your accounts that we should know? For example: Running in EC2 Classic? --->

### References

<!---
Information about referencing Github Issues: https://help.github.com/articles/basic-writing-and-formatting-syntax/#referencing-issues-and-pull-requests

Are there any other GitHub issues (open or closed) or pull requests that should be linked here? Vendor documentation? For example:
--->

* #0000

Oh man. I'm now hitting another annoying error that I know is related to a "working as designed" issue, but partly needs to be made better. I deleted the boot image that the UI made, which was called the same thing as the name in the catalog (CentOS-Stream-8). IIRC, the name locally will just be the name of what it was in the catalog, and you can't change it. So now, since you can't re-create an image with the same name as an old one without waiting some amount of time, I can't test the new TF:

```bash
2023-07-12T16:40:46.223-0700 [ERROR] provider.terraform-provider-ibm_v1.49.0: Response contains error diagnostic: diagnostic_severity=ERROR tf_proto_version=5.3 @caller=github.com/hashicorp/terraform-plugin-go@v0.14.0/tfprotov5/internal/diag/diagnostics.go:55 diagnostic_detail= diagnostic_summary="failed to perform Create Image Operation for cloud instance b4a9a365-9545-4c95-bf5c-d2f2132d4f87 with error  [POST /pcloud/v1/cloud-instances/{cloud_instance_id}/images][400] pcloudCloudinstancesImagesPostBadRequest  &{Code:0 Description:for cloud instance fbd37fc3a30c494c88f9258ff5571cc5, an image with the name CentOS-Stream-8 already exists or may have been recently deleted.  Please wait or reattempt with a unique name Error:bad request Message:}" tf_provider_addr=provider tf_req_id=c26c9a0d-a87c-6e7b-0b09-0ad82225d432 tf_resource_type=ibm_pi_image tf_rpc=ApplyResourceChange @module=sdk.proto timestamp=2023-07-12T16:40:46.223-0700
2023-07-12T16:40:46.244-0700 [DEBUG] State storage *statemgr.Filesystem declined to persist a state snapshot
2023-07-12T16:40:46.244-0700 [ERROR] vertex "ibm_pi_image.image_centos8stream_lon06" error: failed to perform Create Image Operation for cloud instance b4a9a365-9545-4c95-bf5c-d2f2132d4f87 with error  [POST /pcloud/v1/cloud-instances/{cloud_instance_id}/images][400] pcloudCloudinstancesImagesPostBadRequest  &{Code:0 Description:for cloud instance fbd37fc3a30c494c88f9258ff5571cc5, an image with the name CentOS-Stream-8 already exists or may have been recently deleted.  Please wait or reattempt with a unique name Error:bad request Message:}
╷
│ Error: failed to perform Create Image Operation for cloud instance b4a9a365-9545-4c95-bf5c-d2f2132d4f87 with error  [POST /pcloud/v1/cloud-instances/{cloud_instance_id}/images][400] pcloudCloudinstancesImagesPostBadRequest  &{Code:0 Description:for cloud instance fbd37fc3a30c494c88f9258ff5571cc5, an image with the name CentOS-Stream-8 already exists or may have been recently deleted.  Please wait or reattempt with a unique name Error:bad request Message:}
│ 
│   with ibm_pi_image.image_centos8stream_lon06,
│   on main.tf line 38, in resource "ibm_pi_image" "image_centos8stream_lon06":
│   38: resource "ibm_pi_image" "image_centos8stream_lon06" {
│ 
╵
╷
│ Error: failed to provision: failed to Create PVM Instance :[POST /pcloud/v1/cloud-instances/{cloud_instance_id}/pvm-instances][400] pcloudPvminstancesPostBadRequest  &{Code:0 Description:error getting deployment image information: unable to get image f1626ad4-288a-4fc2-ade2-404c387a4fe7/a25cf2e3-6d80-485d-9f8c-2f788caa4c73: Resource not found f1626ad4-288a-4fc2-ade2-404c387a4fe7/a25cf2e3-6d80-485d-9f8c-2f788caa4c73 Error:bad request Message:}
│ 
│   with ibm_pi_instance.centos_instances_lon04[0],
│   on main.tf line 100, in resource "ibm_pi_instance" "centos_instances_lon04":
│  100: resource "ibm_pi_instance" "centos_instances_lon04" {
```

I tried renaming it: 

```hcl
resource "ibm_pi_image" "image_centos8stream_lon04" {
  provider             = ibm.powervslon04
  pi_image_name        = "CentOS-Stream"
  pi_image_id          = local.centos_img_id_lon04
  pi_cloud_instance_id = var.cloud_inst_id_lon04
}
```

but that name seems to be ignored. It's still trying to name it "CentOS-Stream-8", and I keep getting that error.

The only real "bug" here I think is that the name shouldn't be a required parameter since it's ignored in the stock image case.

--- a/website/docs/r/pi_image.html.markdown
+++ b/website/docs/r/pi_image.html.markdown
@@ -11,8 +11,8 @@ description: |-
 Create, update, or delete for a Power Systems Virtual Server image. For more information, about IBM power virtual server cloud, see [getting started with IBM Power Systems Virtual Servers](https://cloud.ibm.com/docs/power-iaas?topic=power-iaas-getting-started).
 
 ## Example usage
-The following example enables you to create a image:
-
+The following example enables you to create a image in your project:
+- stock-images import
 ```terraform
 resource "ibm_pi_image" "testacc_image  "{
   pi_image_name       = "7200-03-02"
@@ -20,7 +20,7 @@ resource "ibm_pi_image" "testacc_image  "{
   pi_cloud_instance_id = "<value of the cloud_instance_id>"
 }
 ```
-
+- COS image import
 ```terraform
 resource "ibm_pi_image" "testacc_image  "{
   pi_image_name       = "test_image"

--- a/website/docs/r/pi_instance.html.markdown
+++ b/website/docs/r/pi_instance.html.markdown
@@ -67,6 +67,7 @@ Review the argument references that you can specify for your resource.
 - `pi_deployment_type` - (Optional, String) Custom deployment type; Allowable value: `EPIC`.
 - `pi_health_status` - (Optional, String) Specifies if Terraform should poll for the health status to be `OK` or `WARNING`. The default value is `OK`.
 - `pi_image_id` - (Required, String) The ID of the image that you want to use for your Power Systems Virtual Server instance. The image determines the operating system that is installed in your instance. To list available images, run the `ibmcloud pi images` command.
+  - **Note**: only images belonging to your project can be used image for deploying a Power Systems Virtual Server instance. To import an images to your project, see [ibm_pi_image](https://registry.terraform.io/providers/IBM-Cloud/ibm/latest/docs/resources/pi_image).
 - `pi_instance_name` - (Required, String) The name of the Power Systems Virtual Server instance. 
 - `pi_key_pair_name` - (Optional, String) The name of the SSH key that you want to use to access your Power Systems Virtual Server instance. The SSH key must be uploaded to IBM Cloud.
 - `pi_license_repository_capacity` - (Optional, Integer) The VTL license repository capacity TB value. Only use with VTL instances. `pi_memory >= 16 + (2 * pi_license_repository_capacity)`.

